### PR TITLE
rename metrics to match prometheus official naming conventions

### PR DESCRIPTION
**Pull request by Shimokuni**
closes https://github.com/discourse/prometheus_exporter/issues/182

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,3 +1,7 @@
+1.0.0 - dd-mm-yyyy
```
Comment (Shimokuni) [inline CHANGELOG:1]: FIX ME

**Review by Shimokuni — COMMENTED** [review]
May I edit [version.rb](lib/prometheus_exporter/version.rb)?

File: README.md [hunk 2]
```diff
@@ -554,14 +561,14 @@ PrometheusExporter::Instrumentation::Resque.start
...
-| ---   | ---                    | ---                                    |
-| Gauge | `processed_jobs_total` | Total number of processed Resque jobs  |
-| Gauge | `failed_jobs_total`    | Total number of failed Resque jobs     |
-| Gauge | `pending_jobs_total`   | Total number of pending Resque jobs    |
-| Gauge | `queues_total`         | Total number of Resque queues          |
-| Gauge | `workers_total`        | Total number of Resque workers running |
-| Gauge | `working_total`        | Total number of Resque workers working |
+| Type  | Name             | Description                            |
+| ---   | ---              | ---                                    |
+| Gauge | `processed_jobs` | Total number of processed Resque jobs  |
```
Comment (SamSaffron) [inline README.md:566]: qw have to fix these name `resque_processed_jobs` etc.

**Comment by SamSaffron** [comment]
thanks, looks like you are making good progress. We do need to also improve metric names and make them more consistent, can you have a look?

File: README.md (hunk 2 above)
Comment (Shimokuni) [reply README.md]: Fixed. https://github.com/discourse/prometheus_exporter/pull/184/commits/ab91609fb46760f4ace046ce68753f379600465a

**Comment by Shimokuni** [comment]
I think this is all fixed now. https://github.com/discourse/prometheus_exporter/pull/184/commits/a845e548b62deffb9f2a61a233f9dca8dcd9a192
And I added some missing tests for unicorn. https://github.com/discourse/prometheus_exporter/pull/184/commits/9669a3b7fb8b44d8cbff1aba2add00751dbd4d9a

---

> We do need to also improve metric names and make them more consistent, can you have a look?

For example, do you mean to fix the inconsistency that resque has a gauge type metric called `resque_failed_jobs`, while sidekiq has a counter type metric called `sidekiq_failed_jobs_total`?

**Comment by SamSaffron** [comment]
This is a very nice change, going to hold off a bit on a release cause it has potential to break quite a few people, let's let it bake for 2-4 weeks.

**Comment by salzig** [comment]
Would've been awesome if changes like those get a graceperiod where both names are provided. We ended up in a situation where our dashboards are working for one or the other environemnt (as in stage/production), cause our deployment cycles are sadly not as fast as we wish for (Customer does it's own QA additionally, and very slow).

**Comment by SamSaffron** [comment]
@salzig kind of impossible to do this without some very bad side effects (exploding metric counts) 

The change was flagged as a major change we did major version bump here.

**Comment by salzig** [comment]
@SamSaffron i don't see a problem with the name change in the major version. Maybe our setup is "strange", but name changes on one day to another are quite annoying, as we have to choose between showing the old or the new name, or query both metrics in our panels, as our Grafana installation is central across all environments. 

Sorry, overall my last message was written in frustration, cause another metric (http request duration) was also renamed, and additionally lost a label which we used (status). Keeping a old version, isn't really an option. So we need to figure out how to proceed. 

Thanks to all of you for providing and maintaining this as open source.

**Pull request merged**